### ISSUE TITLE
WMVs are silently skipped for some wine games

### ISSUE THREAD

**Issue by CartoonFan** [Tk-Glitch/PKGBUILDS#503]
Hello! It seems that .wmv files (Windows Media) do not currently play properly with wine. I was formerly able to able to fix this by using "winetricks wmp9 devenum quartz", and possibly adding amstream, but now...nothing. I have tested three games so far:  One Piece Pirate Warriors 3, Tokyo Xanadu eX+, and [the demo for Pirates of Black Cove](https://www.fileplanet.com/archive/p-345/Pirates-of-Black-Cove-Demo). With native quartz, all three silently skip wmv files when they are encountered (although the demo will crash with some combination of native/builtin dlls), and with builtin quartz, there is usually either stuttering or a crash. Additionally, both ffplay and gst-play-1.0 seem to be able to play the files, and changing the wine version through lutris doesn't seem to help.

I've been investigating this issue, but I'm not sure what to do. If it's relevant, I'm using a 32-bit prefix and gst-libav, although it doesn't seem like gstreamer is even being used for these programs. My wine version is wine-5.2.r5.g469cbe7e ( TkG Staging Esync Fsync Vkd3d ) and I'm using gst-libav-tkg. Thanks!

**Comment by ViNi-Arco** [comment]
This is a known "problem" and so far has not had a plausible solution, see how it is discussed:

https://bugs.winehq.org/show_bug.cgi?id=9127
https://appdb.winehq.org/viewbugs.php?iBugId=9127
https://github.com/ValveSoftware/Proton/issues/1464

**Comment by CartoonFan** [comment]
Indeed. In fact, over the last couple days I have been trying things from https://github.com/Winetricks/winetricks/issues/1087#issuecomment-583763440 and https://github.com/Winetricks/winetricks/pull/1489 without much success. I've ran into the problem before, and usually the solution is to install some combination of wmp9, quartz, devenum, amstream, or ffdshow, but wine not playing the videos at all after doing so is new to me. The WineHQ issues you have there seem to be more focused on MPEG/gstreamer issues, like those in some Ys games (Origin, Oath in Felgana), which may be worked around by using ffdshow and such. For this issue, I'm not even sure if gstreamer is being used, since it doesn't show up at all in the debug log I made.

**Comment by ViNi-Arco** [comment]
From what I have read it seems that gstreamer is disabled, and ffdshow is no longer a solution, GloriousEggroll's Proton seems to have an alternative solution, as there are games that Valve's Wine and Proton cannot reproduce an image, but GloriousEggroll's Proton can.

**Comment by ViNi-Arco** [comment]
It was painful to do the NFS Carbon campaign without the cutscene, and not having the classic introductions of GTA SA, GTA III and GTA Vice City, I hope that some light will appear at the end of the tunnel, as it is difficult, the alternatives we had before most doesn't work anymore..

**Comment by CartoonFan** [comment]
I agree. I'm also looking at the GloriousEggroll fork you mentioned (https://github.com/GloriousEggroll/proton-ge-custom), and I plan to try it out soon. It would be nice if I can get it to work with proton-tkg, but if not, I'll just have to work a little harder. Thanks for your help!

**Comment by ViNi-Arco** [comment]
There is a PC version of NFS Carbon, and it works almost perfect, only cutscenes don't work like I told you, see it working:

![2020-01-08-200827_1440x900_scrot](https://user-images.githubusercontent.com/57958460/75129157-7d010480-56a6-11ea-8c6c-1c831737cf03.png)

I have been testing GTA Vice City and San Andreas with DXVK these days and they work very well, they just don't show the introduction for me, GTA III is working very well with DXVK these days, but d3d8to9 is necessary for Vice City and III work with DXVK https://github.com/crosire/d3d8to9

**Comment by ViNi-Arco** [comment]
You said that GTA III was crashing for you, how about trying again, it’s working fine for me, see it working:

![2020-02-24-014754_1440x900_scrot](https://user-images.githubusercontent.com/57958460/75129690-e550e580-56a8-11ea-9e1d-8eb85f9b0232.png)

**Comment by z0z0z** [comment]
There's been many commits in upstream Wine recently regarding quartz/wmp/winegstreamer.

It may be that in the process of trying to get native support, it broke these DLL workarounds for some games.

**Comment by CartoonFan** [comment]
@z0z0z  I see. I felt like a Wine thing, but going back to earlier wine versions didn't seem to fix it, so I wasn't sure.

@ViNi-Arco Good call with GTA III. It seems to be working somewhat (intro included) with Proton-TKG 5.2 r4. I haven't tried d3d8to9 on it yet, though.

**Comment by Tk-Glitch** [comment]
Proton -GE and -tkg are sharing the same patches regarding codecs handling.
A couple things that could lead to a different behavior would be protonfixes (a tool I highly dislike that comes bundled with -GE that installs all kinds of undesirable stuff in your game prefixes on launch) and the fact that -GE ships with a huge list of libs while -tkg is only targeting system libs (at least currently).
If you got all the optional dependencies installed, the latter shouldn't make a difference.

Regarding the actual issue here, it might be worth trying the same workarounds on a 32-bit only prefix (if that's not something you have tried already) as it was still a requirement to get wmv fmvs to work on the couple games I'm playing that have some.

**Comment by CartoonFan** [comment]
@Tk-Glitch Thanks for the info. I've tried protonfixes in the past, but since proton and winetricks are working better together these days, I'm not using it right now. I have also tried a 32-bit prefix, as well as cleaning caches, old prefixes, etc. with no change in behavior. I'm not sure if there's broken libs or something even stranger at this point, but I guess I'll just wait and try whatever I can.

**Comment by robzombie91** [comment]
This may be a long shot but try installing lav-filters (https://github.com/Nevcairiel/LAVFilters/releases) in your prefix. It has fixed at least one game for me (zero escape the nonary games, due to it playing mplayer videos before the game starts though the audio is still broken) but do it at your own risk.

**Comment by CartoonFan** [comment]
@robzombie91 Still skipping with lav-filters, unfortunately. I'm thinking it might be something wrong with my system. The one interesting thing I've noticed is that builtin quartz crashes Tokyo Xanadu eX+ with my wine-tkg (wine-5.2.r6.gd68ab574 ( TkG Staging Esync Fsync Vkd3d )) but not with lutris-5.2, so maybe it is partially wine-related :confused:

**Comment by robzombie91** [comment]
like i said, long shot. it worked for me but it might not work for everyone

**Comment by CartoonFan** [comment]
So, after a lot of testing, it seems that the solution includes the things that have been mentioned so far. Currently for 5.3, Tokyo Xanadu eX+ works with a semi-complex setup. Native qasf, quartz, and wmp9 dlls need to be used, and devenum, dsound, and qedit have to be set to builtin. Additionally, xact needs to be installed to avoid crashes. After that, I used lav-filters (which needs native oleaut32), and set wmvs to play with LAV's filters. With all that done, videos play correctly.

So far, Pirate Warriors 3's videos aren't playing when using roughly the same setup, as Proton seems to be loading almost everything as builtin dlls. So, I'm stuck again until I find something that works. Still, this is a big step forward in terms of usability. Thanks for all your support thus far, and I apologize if I seem like I'm not giving your suggestions a fair chance. The process of testing and refining things until it works is very time consuming, so I might give up on something, only to come back to it later, like with lav-filters.

**Comment by random2324** [comment]
There seems to be now some wine patches for this issue: https://github.com/Guy1524/wine/commits/mfplat

Maybe those can be included?

**Comment by Tk-Glitch** [comment]
Too early 🐸

**Comment by ViNi-Arco** [comment]
@random2324  I've been following the work that @Guy1524 has been doing, and it's already a start! `Guy1524` deserves congratulations for the result, really some games that I tested are working the cinematics now.

@CartoonFan In the last release of Wine, the cutscenes of `NFS: Underground 2` worked now, later I send a printscreen to demonstrate it working.

For more information of what is happening, I recommend keeping an eye here for now:
https://github.com/Guy1524/wine
https://github.com/ValveSoftware/Proton/issues/1464

**Comment by CartoonFan** [comment]
Awesome! Thanks @ViNi-Arco!

**Comment by ViNi-Arco** [comment]
I'm sorry @CartoonFan, I won't be able to send screenshots of the scenes from NFS: Underground 2, my HDD is defective, and I'm trying to recover my data, it will take a while for me to buy a new HDD :disappointed:

**Comment by CartoonFan** [comment]
Sorry to hear that, @ViNi-Arco. Hopefully, you'll be able to start gaming again soon. On the brighter side, hopefully the media work will also be done by the time you get your new HDD.

**Comment by CartoonFan** [comment]
It seems like movie playback has been narrowed down, at least for me, to native qasf and wmvcore, along with LAV-Filters. I tried the MFPlat patches, but even with those, the results seem to be the same, so maybe my case isn't that related to MFPlat. I'll guess I'll just keep waiting for a fix, though.

**Comment by Tk-Glitch** [comment]
Just so you guys are aware, Derek is on some kind of vacation regarding MFPlat right now while he's back on EAC-related work.

**Comment by ViNi-Arco** [comment]
> Hopefully, you'll be able to start gaming again soon. On the brighter side, hopefully the media work will also be done by the time you get your new HDD.

Hi @CartoonFan , thank you very much for your words, finally getting my new Hard Disk, he was almost 3 months stopped, the company where I bought it had stopped because of this damn virus.

> It seems like movie playback has been narrowed down, at least for me, to native qasf and wmvcore, along with LAV-Filters. I tried the MFPlat patches, but even with those, the results seem to be the same, so maybe my case isn't that related to MFPlat.

I am sad to hear that MFPlat patches did not help you much.

with my new Hard Disk in hand I was finally able to test if the game cutscene is working, as I have a new OS installation, and I cannot recover all data from my defective HDD, I have few games to test, with `MFPlat patches` Dark Souls Remastered is working, the Cutscene for me before was just a black screen with the background audio, see:
https://img.tpx.cz/uploads/Dark%20Souls%20Remastered%20Intro%20Cutscene%20print01.png
https://img.tpx.cz/uploads/Dark%20Souls%20Remastered%20Intro%20Cutscene%20print02.png

I installed NFS Underground 2 as I have the DVD I was able to test it, and as I promised in my previous message here the images of the Cutscene working:
https://img.tpx.cz/uploads/NFS%20Undergound%202%20Intro%20Cutscene%20Print01.png
https://img.tpx.cz/uploads/NFS%20Undergound%202%20Intro%20Cutscene%20Print02.png

Gradually as I download more games on Steam, I will test to see if `MFPlat` helps with more games.

**Comment by ViNi-Arco** [comment]
> Just so you guys are aware, Derek is on some kind of vacation regarding MFPlat right now while he's back on EAC-related work.

Hello Etienne, how have you been? wow you gave a nice makeover in your repository, I just looked once and for all through the CellPhone, things advanced a lot with the time I was away, anyway Derek, you and the community did a great job

**Comment by CartoonFan** [comment]
I've got a small status update (and a bit of a quirk) with One Piece's WMVs. Maybe's it's due to something I installed (or something from https://github.com/ValveSoftware/Proton/issues/1464), but Proton is now playing Pirate Warriors' WMVs! However, there seems to be an issue with the colors:

Proton:
![Screenshot_2020-07-26_23-04-22](https://user-images.githubusercontent.com/29664894/88761602-4bdca280-d124-11ea-9d3d-39668b99a167.png)

Video Player:
![Screenshot_2020-07-28_22-51-24](https://user-images.githubusercontent.com/29664894/88761928-f48b0200-d124-11ea-887d-1cbd8d7aa453.png)

However, this is a pretty big step forward, as files will now play on a clean Proton wine prefix. I just wanted to throw this out there in case someone is curious or knows of a solution. Thanks!

**Comment by ViNi-Arco** [comment]
@CartoonFan RE REVELATIONS 2 that uses video in WMV does not work without installing WMP11 with Protontricks:
https://www.protondb.com/app/287290

which version is exactly WMV from One Piece? RE REVELATIONS 2 is WMV3
![rerevelationcodec](https://user-images.githubusercontent.com/57958460/89747416-17a6a180-da95-11ea-8c26-0c7b0dd1bcd3.png)

**Comment by CartoonFan** [comment]
It seems like it's using WMV3:
![Screenshot_2020-08-09_22-02-44](https://user-images.githubusercontent.com/29664894/89757145-4888d580-da99-11ea-8382-ead8f30c9ee7.png)

Also, installing wmp11 unfortunately didn't seem to fix the green color issue. Additionally, wmp11's video playback appears slower than whatever native playback is doing.
